XenModel/XenAPI/HTTP.cs

```diff
@@ -771,16 +771,20 @@ public static Stream HttpGetStream(Uri uri, IWebProxy proxy, int timeoutMs)
         public static void Get(DataCopiedDelegate dataCopiedDelegate, FuncBool cancellingDelegate,
             Uri uri, IWebProxy proxy, string path, int timeoutMs)
         {
-            var dir = Path.GetDirectoryName(path);
-            if (dir == null)
-            {
-                if (path == null)
-                    throw new ArgumentNullException(nameof(path));
+            if (string.IsNullOrWhiteSpace(path))
                 throw new ArgumentException(nameof(path));
-            }
 
-            var tmpFile = Path.Combine(dir, Path.GetRandomFileName());
-            File.Delete(tmpFile);
+            var tmpFile = Path.GetTempFileName();
+
+            if (Path.GetPathRoot(path) != Path.GetPathRoot(tmpFile))

```

Comment:
I'd add a comment to show why we're doing this. Might not be very clear on why we did this in a few years